lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_localize_helper.rb

```diff
@@ -171,6 +172,79 @@ def self.verify_pr_diff(main, library, main_strings, lib_strings, source_diff)
             end
           end
         end
+
+        ########
+        # @group Downloading translations from GlotPress
+        ########
+
+        # Create the `available_languages.xml` file.
+        #
+        # @param [String] res_dir The relative path to the `…/src/main/res` directory
+        # @param [Array<String>] locale_codes The list of locale codes to include in the available_languages.xml file
+        #
+        def self.create_available_languages_file(res_dir:, locale_codes:)
+          doc = Nokogiri::XML::Builder.new(encoding: 'UTF-8') do |xml|
+            xml.comment('Warning: Auto-generated file, do not edit.')
+            xml.resources do
+              xml.send(:'string-array', name: 'available_languages', translatable: 'false') do
+                locale_codes.each { |code| xml.item(code.gsub('-r', '_')) }
+              end
+            end
+          end
+          File.write(File.join(res_dir, 'values', 'available_languages.xml'), doc.to_xml)
+        end
+
+        # Perform some quick basic checks about an individual `<string>` tag and print warnings accordingly
+        #
+        # @param [Nokogiri::XML::Node] string_tag The XML tag/node to check
+        #
+        def self.quick_lint(string_tag)
+          if string_tag['formatted'] == 'false' && string_tag.content.include?('%%')
+            UI.important "Warning: translation for '#{string_tag['name']}' has attribute formatted=false, but still contains escaped '%%' in translation."
+          end
+        end
+        private_class_method :quick_lint
+
+        # Download translations from GlotPress
+        #
+        # @param [String] res_dir The relative path to the `…/src/main/res` directory
+        # @param [String] glotpress_project_url The base URL to the glotpress project to download the strings from
+        # @param [Array<Hash{Symbol=>String}>] locales_map An array of locales to download. Each item in the array must be a
+        #                                      Hash with keys `:glotpress` and `:android` containing the respective locale codes.
+        #
+        def self.download_from_glotpress(res_dir:, glotpress_project_url:, locales_map:)
```

Comment:
It may or may not make sense to abstract this for use with iOS, though I could see building the Android-specific implementation now and doing that later too. Just figured I'd flag it in case it affects other decisions

Reply:
Right, but the only thing that would be common when porting this to iOS would be the method signature, because implementation wise, that will be quite different for iOS (as it would download `.strings` Plist file not an XML file, the post-processing of XML to recopy the attributes would not make sense on iOS either, and the location at which the files would have to be written (as well as the `Localizable.strings` vs `strings.xml` file names) would also differ.

Actually, not even the signature would be exactly the same, as `res_dir` would have a slightly different meaning for iOS (it's not a standard location like `src/main/res` in Android, as in iOS you can put your `.lproj` folders anywhere on disk as long as they are properly configured in your xcodeproj), so even the signature of this method might differ in iOS… So the only line from the code that might be common is `uri.open` (… and even the URL for that one will need a different `format=` param) 🤷